feat: rename `id` property to `exporter` in CIRCULAR_REEXPORT error

sapphi-red opened the pull request:
Rename `id` property to `exporter` in CIRCULAR_REEXPORT error to align with Rollup.

sapphi-red commented:
* **#7593** <a href="https://app.graphite.com/github/pr/rolldown/rolldown/7593?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7592** <a href="https://app.graphite.com/github/pr/rolldown/rolldown/7592?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/rolldown/rolldown/7592?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#7591** <a href="https://app.graphite.com/github/pr/rolldown/rolldown/7591?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite: merge</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/rolldown?ref=merge-queue-instructions-in-stack-comment&prId=7592).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

hyf0 commented on crates/rolldown/tests/snapshots/integration_test262__test262_module_code.snap:
```diff
@@ -796,17 +796,17 @@ Cannot assign to import 'B'
-'x' cannot be exported from 'instn-iee-err-circular.js' as it is a reexport that references itself.
-'x' cannot be exported from 'instn-iee-err-circular_FIXTURE.js' as it is a reexport that references itself.
-'x' cannot be exported from 'instn-iee-err-circular_FIXTURE.js' as it is a reexport that references itself.
+'x' cannot be exported from <test262>/test/language/module-code/instn-iee-err-circular.js' as it is a reexport that references itself.
```
I keep thinking if we could just use the absolute id for all errors.

sapphi-red replied:
This diff wasn't expected. I've changed the test script to pass `cwd` properly.

sapphi-red committed:
feat: rename `id` property to `exporter` in CIRCULAR_REEXPORT error (#7592)

Rename `id` property to `exporter` in CIRCULAR_REEXPORT error to align with Rollup.
